### PR TITLE
[FIX] ensure inverse transform accepts array-like

### LINKED ISSUE #5969: [ENH] support for array-like as input to inverse_transform
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe your proposed enhancement in detail.

sklearn inverse_transform methods in general can accept array-like inputs not just numpy array.

```python
import numpy as np
from scipy import signal
import pandas as pd
from sklearn.decomposition import FastICA
import polars

# generate data
np.random.seed(0)
n_samples = 2000
time = np.linspace(0, 8, n_samples)
s1 = np.sin(2 * time)  # Signal 1 : sinusoidal signal
s2 = np.sign(np.sin(3 * time))  # Signal 2 : square signal
s3 = signal.sawtooth(2 * np.pi * time)  # Signal 3: saw tooth signal
S = np.c_[s1, s2, s3]
S += 0.2 * np.random.normal(size=S.shape)  # Add noise
S /= S.std(axis=0)
A = np.array([[1, 1, 1], [0.5, 2, 1.0], [1.5, 1.0, 2.0]]) 
X = np.dot(S, A.T)

ica = FastICA(n_components=3, whiten="arbitrary-variance")

S_ = ica.fit_transform(X)

ica.inverse_transform(S_)
ica.inverse_transform(pd.DataFrame(S_))
ica.inverse_transform(polars.DataFrame(S_))
```

That includes pandas and polars dataframes.

Currently nilearn most inverse_transform methods (not checked them all) only accept numpy arrays: or at least do not accept pandas or polars dataframe.

Since the last release, many of the Nilearn estimator with a transform method can also output to pandas or polars (see https://github.com/nilearn/nilearn/pull/5672). But that brings us to a situation where somes estimators can transform to a type that cannot be used by inverse_transform.

### Benefits to the change

- behavior match that of sklearn
- internal consistency

### Pseudocode for the new behavior, if applicable

code to show expected behavior and some current failures and bug

```python
from nilearn.maskers import SurfaceMasker, NiftiMasker
from nilearn.conftest import _img_3d_mni, _make_surface_img, _surf_img_1d
import pandas as pd
import polars
import warnings

for masker, input in zip(
    [NiftiMasker(), SurfaceMasker(), SurfaceMasker()],
    [_img_3d_mni(), _surf_img_1d(), _make_surface_img(1)],
):
    for output in ["pandas", "polars", "default"]:

        masker.set_output(transform=output)

        masker.fit(input)
        
        try:
            signal = masker.transform(input)
        except Exception as e:
            warnings.warn(
                f"\nCannot transform for {masker.__class__.__name__} with {input=}"
                f"and {output=}: {str(e)}\n",
                stacklevel=2,
            )
            continue


        # Check internal consistency:
        # check we can do masker.inverse_transform(masker.transform(input))
        try:
            masker.inverse_transform(signal)
        except Exception as e:
            warnings.warn(
                f"\nCannot inverse_transform for {masker.__class__.__name__} "
                f"with input {type(signal)=} and {output=}: {str(e)}\n",
                stacklevel=2,
            )

        if output == "default":
            for s in [pd.DataFrame(signal), polars.DataFrame(signal)]:
                try:
                    masker.inverse_transform(s)
                except Exception as e:
                    warnings.warn(
                        f"\nCannot inverse_transform for {masker.__class__.__name__} "
                        f"with input {s.__class__.__name__} and {output=}: {str(e)}\n",
                        stacklevel=2,
                    )
        # TODO check that when output is pandas or polars, we can still pass other array like
```

output

```
sys:1: UserWarning: 
Cannot transform for SurfaceMasker with input=<SurfaceImage (9,)>and output='pandas': Length mismatch: Expected axis has 1 elements, new values have 9 elements

sys:1: UserWarning: 
Cannot transform for SurfaceMasker with input=<SurfaceImage (9,)>and output='polars': dimensions of `schema` (9) must match data dimensions (1)

sys:1: UserWarning: 
Cannot inverse_transform for SurfaceMasker with input DataFrame and output='default': Input to 'inverse_transform' has wrong shape.
Last dimension should be 9.
Got 1.

sys:1: UserWarning: 
Cannot inverse_transform for SurfaceMasker with input DataFrame and output='default': 'DataFrame' object has no attribute 'ndim'

sys:1: UserWarning: 
Cannot inverse_transform for SurfaceMasker with input type(signal)=<class 'polars.dataframe.frame.DataFrame'> and output='polars': 'DataFrame' object has no attribute 'ndim'

sys:1: UserWarning: 
Cannot inverse_transform for SurfaceMasker with input DataFrame and output='default': 'DataFrame' object has no attribute 'ndim'
```

note that:
- niftimasker seems to work fine
- surface masker not being able with 1D surface image when output is polars or pandas is actually a bug

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #5969

Changes proposed in this pull request:

- [x] add regression tests
- [x] fix
- [ ] changelog

### CODE REVIEW

File: nilearn/_utils/estimator_checks.py [hunk 1]
```diff
@@ -1262,51 +1290,6 @@ def check_nilearn_methods_sample_order_invariance(estimator_orig):
             )
 
 
-@ignore_warnings()
-def check_estimator_set_output(estimator_orig):
```
Comment (Remi-Gau) [inline nilearn/_utils/estimator_checks.py]: this was duplicated

File: nilearn/maskers/surface_labels_masker.py [hunk 2]
```diff
@@ -455,10 +455,10 @@ def inverse_transform(self, signals):
         """
         check_is_fitted(self)
 
-        return_1D = signals.ndim < 2
-
         signals = self._check_array(signals)
 
+        return_1D = signals.ndim < 2
```
Comment (Remi-Gau) [inline nilearn/maskers/surface_labels_masker.py:440]: _check_array will cast to numpy: so avoid errors of dataframe not having a dim

File: nilearn/_utils/estimator_checks.py [hunk 3]
```diff
@@ -849,38 +859,70 @@ def check_set_output(estimator_orig):
...
 
+        assert isinstance(signal, expected_type)
+
+        if hasattr(estimator, "inverse_transform"):
+            for v in to_inverse_transform.values():
+                estimator.inverse_transform(v)
+
     # check on 1D image for estimators that accepts surface
+    # TODO errors transforming 1D images
+    # when output set to "pandas" or "polars"
```
Comment (Remi-Gau) [inline nilearn/_utils/estimator_checks.py:873]: was not able to fix this : would prefer to open an issue and tackle in a follow up

Comment (bthirion) [reply nilearn/_utils/estimator_checks.py:873]: Yep, makes sense
